subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c

```diff
@@ -0,0 +1,251 @@
+	if (nrfx_gpiote_channel_alloc(&md_gpiote, &md_gpiote_channel) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+
+	mpsl_fem_nrf2240_interface_config_t cfg = {
+		.fem_config = {
+			.output_power_dbm = FEM_OUTPUT_POWER_DBM,
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), pwrmd_gpios)
+			.output_power_alt_dbm = FEM_OUTPUT_POWER_ALT_DBM,
+#else
+			.output_power_alt_dbm = FEM_OUTPUT_POWER_DBM,
+#endif
+			.bypass_gain_db = DT_PROP(DT_NODELABEL(nrf_radio_fem), bypass_gain_db),
+		},
+		.cs_pin_config = {
+			.gpio_pin      = {
+				.p_port   = MPSL_FEM_GPIO_PORT_REG(cs_gpios),
+				.port_no  = MPSL_FEM_GPIO_PORT_NO(cs_gpios),
+				.port_pin = MPSL_FEM_GPIO_PIN_NO(cs_gpios),
+			},
+			.enable        = true,
+			.active_high   = MPSL_FEM_GPIO_POLARITY_GET(cs_gpios),
+			.gpiote_ch_id  = cs_gpiote_channel
+		},
+		.md_pin_config = {
+			.gpio_pin      = {
+				.p_port   = MPSL_FEM_GPIO_PORT_REG(md_gpios),
+				.port_no  = MPSL_FEM_GPIO_PORT_NO(md_gpios),
+				.port_pin = MPSL_FEM_GPIO_PIN_NO(md_gpios),
+			},
+			.enable        = true,
+			.active_high   = MPSL_FEM_GPIO_POLARITY_GET(md_gpios),
+			.gpiote_ch_id  = md_gpiote_channel
+		},
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), pwrmd_gpios)
+		.pwrmd_pin_config = {
+			.gpio_pin      = {
+				.p_port   = MPSL_FEM_GPIO_PORT_REG(pwrmd_gpios),
+				.port_no  = MPSL_FEM_GPIO_PORT_NO(pwrmd_gpios),
+				.port_pin = MPSL_FEM_GPIO_PIN_NO(pwrmd_gpios),
+			},
+			.enable        = true,
+			.active_high   = MPSL_FEM_GPIO_POLARITY_GET(pwrmd_gpios),
+		}
+#endif
+	};
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), twi_if)
+	fem_nrf2240_twi_configure(&cfg);
+#else
+	fem_nrf2240_twi_configure_disabled(&cfg);
+#endif
+
+#if DT_NODE_HAS_PROP(MPSL_FEM_TWI_IF, init_regs)
+	const int32_t fem_twi_init_regs[] = DT_PROP(MPSL_FEM_TWI_IF, init_regs);
+	const uint8_t nb_regs = ARRAY_SIZE(fem_twi_init_regs) / 2;
+
+	BUILD_ASSERT(ARRAY_SIZE(fem_twi_init_regs) % 2 == 0,
+		"The number of specified nRF2240 TWI initialization registers to write differs from"
+		" the number of values specified for those registers.");
+
+	mpsl_fem_twi_reg_val_t init_regs[nb_regs];
+
+	cfg.twi_regs_init_map.p_regs = init_regs;
+	cfg.twi_regs_init_map.nb_regs = nb_regs;
+
+	fem_nrf2240_twi_init_regs_configure(&cfg);
+#endif /* DT_NODE_HAS_PROP(MPSL_FEM_TWI_IF, init_regs) */
+
+	IF_ENABLED(CONFIG_HAS_HW_NRF_PPI,
+		   (err = mpsl_fem_utils_ppi_channel_alloc(cfg.ppi_channels,
+					ARRAY_SIZE(cfg.ppi_channels));));
+	IF_ENABLED(CONFIG_HAS_HW_NRF_DPPIC,
+		   (err = mpsl_fem_utils_ppi_channel_alloc(cfg.dppi_channels,
+					ARRAY_SIZE(cfg.dppi_channels));));
+	if (err) {
+		return err;
+	}
+
+	err = mpsl_fem_nrf2240_interface_config_set(&cfg);
+
+	nrfx_twim_uninit(&m_twim);
+
+	return err;
+}
+
+static int mpsl_fem_init(void)
+{
+	mpsl_fem_device_config_254_apply_set(IS_ENABLED(CONFIG_MPSL_FEM_DEVICE_CONFIG_254));
+
+	return fem_nrf2240_configure();
+}
+
+SYS_INIT(mpsl_fem_init, POST_KERNEL, 49);
```

Comment:
Applies both here and for nRF2240: why this particular priority? It looks very specific

Reply:
Leftover related to possibility to use CONFIG_I2C_SHELL, I will come with solution.

Reply:
Solved in e68e23394d0f272f5d54059d051dc14fd4903d7a